src/python/pants/reporting/streaming_workunit_handler.py

```diff
@@ -3,11 +3,25 @@
 
 import threading
 from contextlib import contextmanager
-from typing import Any, Callable, Iterable, Iterator, Optional
+from dataclasses import dataclass
+from typing import Any, Callable, Iterable, Iterator, List, Optional
 
+from pants.engine.fs import Digest
+from pants.engine.internals.scheduler import SchedulerSession
 from pants.util.logging import LogLevel
 
 
+@dataclass(frozen=True)
+class StreamingWorkunitContext:
+    _scheduler: SchedulerSession
+
+    def digests_to_bytes(self, digests: List[Digest]) -> List[bytes]:
+        return self._scheduler.digests_to_bytes(digests)
+
+    def ensure_remote_has_recursive(self, digests: List[Digest]) -> None:
```

Comment:
not sure what this API is for, maybe add a docstring ?
also accept Sequence[Digest]